packages/core/src/oidc/grants/refresh-token.ts

```diff
@@ -246,6 +246,17 @@ export const buildHandler: (
       error.statusCode = 403;
       throw error;
     }
+
+    // Check if the organization requires MFA and the user has MFA enabled
+    const { isMfaRequired, hasMfaConfigured } = await queries.organizations.getMfaData(
+      organizationId,
+      account.accountId
+    );
+    if (isMfaRequired && !hasMfaConfigured) {
+      const error = new AccessDenied('organization requires MFA but user has no MFA configured');
+      error.statusCode = 403;
+      throw error;
+    }
```

Comment:
Should we also revoke the current refresh token and session?  e.g. Calling `libraries/user.ts#signoutUser`

Reply:
i don't think so. a user should be available to add MFA and come back to fetch the token afterwards